internal/discovery/cloud.go

```diff
@@ -36,14 +37,14 @@ func (d CloudDiscovery) GetInterval() time.Duration {
 	return d.interval
 }
 
-func (d CloudDiscovery) Discover() (string, error) {
-	client := &http.Client{Transport: &http.Transport{Proxy: nil}}
-	cloudData, err := cloud.NewCloudInstance(utils.Executor{}, client)
+func (d CloudDiscovery) Discover(ctx context.Context) (string, error) {
```

Comment:
Just a comment, don't change it XD
It would've been nice to have something like `Discover` and `DiscoverWithContext`, `Publish` and `PublishWithContext`, it looks much more golang like hehe

Reply:
I was thinking about that, but we need two interfaces, or two methods on the same interface, in the end the correct way is with context, at least in this case, is a loop started in a goroutine that already has context (in the agent file)